deploy/examples/create-external-cluster-resources.py

```diff
@@ -565,31 +542,113 @@ def create_cephCSIKeyring_cephFSNode(self):
-                            "entity": entity,
-                            "caps": ["mon", "allow r, allow command 'osd blocklist'",
-                                     "mgr", "allow rw",
-                                     "osd", "allow rw tag cephfs *=*",
-                                     "mds", "allow rw"],
-                            "format": "json"}
+                caps = {"mon": "allow r, allow command 'osd blocklist'",
+                        "mgr": "allow rw",
+                        "osd": "allow rw tag cephfs *=*",
+                        "mds": "allow rw"}
             else:
                 entity = "client.csi-cephfs-node-{}-{}".format(cluster_name,cephfs_filesystem)
-                cmd_json = {"prefix": "auth get-or-create",
-                            "entity": entity,
-                            "caps": ["mon", "allow r, allow command 'osd blocklist'",
-                                     "mgr", "allow rw",
-                                     "osd", "allow rw tag cephfs data={}".format(
-                                        cephfs_filesystem),
-                                     "mds", "allow rw"],
-                            "format": "json"}
+                caps = {"mon": "allow r, allow command 'osd blocklist'",
+                        "mgr": "allow rw",
+                        "osd": "allow rw tag cephfs data={}".format(cephfs_filesystem),
+                        "mds": "allow rw"}
+        else:
+            caps = {"mon": "allow r, allow command 'osd blocklist'",
+                    "mgr": "allow rw",
+                    "osd": "allow rw tag cephfs *=*",
+                    "mds": "allow rw"}
+            
+        return caps,entity    
+    
+    def get_rbd_provisioner_caps_and_entity(self):
+        entity = "client.csi-rbd-provisioner"
+        caps = {}
+        if self._arg_parser.restricted_auth_permission:
+            rbd_pool_name = self._arg_parser.rbd_data_pool_name
+            cluster_name = self._arg_parser.cluster_name
+            rados_namespace = self._arg_parser.rados_namespace
+            if rbd_pool_name == "" or cluster_name == "" or rados_namespace == "":
+                raise ExecutionFailureException(
+                    "mandatory flags not found, please set the '--rbd-data-pool-name', '--cluster-name' and --rados-namespace flags")
+            entity = "client.csi-rbd-provisioner-{}-{}-{}".format(cluster_name,rbd_pool_name,rados_namespace)
+            caps = {"mon": "profile rbd, allow command 'osd blocklist'",
+                    "mgr": "allow rw",
+                    "osd": "profile rbd pool={}".format(rbd_pool_name)}
         else:
-            cmd_json = {"prefix": "auth get-or-create",
+            caps = {"mon": "profile rbd, allow command 'osd blocklist'",
+                    "mgr": "allow rw",
+                    "osd": "profile rbd"}
+        
+        return caps,entity    
+    
+    def get_rbd_node_caps_and_entity(self):
```

Comment:
```suggestion
    def get_rbd_node_caps_and_entity(self):
        entity = "client.csi-rbd-node"
        caps = {"mon": "profile rbd, allow command 'osd blocklist'",
                "osd": "profile rbd"}
        if self._arg_parser.restricted_auth_permission:
            rbd_pool_name = self._arg_parser.rbd_data_pool_name
            cluster_name = self._arg_parser.cluster_name
            rados_namespace = self._arg_parser.rados_namespace
            if rbd_pool_name == "" or cluster_name == "" or rados_namespace == "":
                raise ExecutionFailureException(
                    "mandatory flags not found, please set the '--rbd-data-pool-name', '--cluster-name' and --rados-namespace flags")
            entity = "{}-{}-{}-{}".format(entity, cluster_name, rbd_pool_name, rados_namespace)
            caps["osd"] = "profile rbd pool={}".format(rbd_pool_name)
        
        return caps,entity
```

Reply:
Optimized all the function's, Thanks for reviewing:)